src/main/java/org/jabref/gui/dialogs/AutosaveUIManager.java

```diff
@@ -25,10 +29,20 @@ public AutosaveUIManager(BasePanel panel) {
 
     @Subscribe
     public void listen(@SuppressWarnings("unused") AutosaveEvent event) {
-        try {
-            new SaveDatabaseAction(panel, Globals.prefs, Globals.entryTypesManager).save(SaveDatabaseAction.SaveDatabaseMode.SILENT);
-        } catch (Throwable e) {
-            LOGGER.error("Problem occurred while saving.", e);
+        if (timer != null) {
+            timer.cancel();
         }
+
+        timer = new Timer();
+        timer.schedule(new TimerTask() {
```

Comment:
Wild guess: Can this be written as lambda?

Reply:
I also thought so but unfortunately not. See [here](https://stackoverflow.com/a/37970821)